src/justenoughicu.c

```diff
@@ -28,24 +29,32 @@ int icu_case(lua_State *L) {
   u_strFromUTF8(input_as_uchar, l, &l, input, input_l, &err);
 
   /* Now do the conversion */
-  conversion_function_t conversion;
   UChar *output;
   int32_t l2 = 0;
 
-  if (strcmp(recase, "upper") == 0) {
-    conversion = u_strToUpper;
-  } else if (strcmp(recase, "lower") == 0) {
-    conversion = u_strToLower;
-  // } else if (strcmp(recase, "title") == 0) {
-    // conversion = u_strToTitle;
+  if (strcmp(recase, "title") == 0) {
+    conversion_function_ti conversion;
+    UBreakIterator *iter;
+    conversion = u_strToTitle;
+    l2 = conversion(NULL, 0, input_as_uchar, l, iter, locale, &err);
```

Comment:
Instead of passing in `0` here, use a small stack buffer if you think the results will fit in some cases. Then you can avoid the malloc and second trip through the conversion function. See [this sample](http://userguide.icu-project.org/conversion/converters#TOC-4.-Pre-flighting) . This goes for the other examples.